### PR TITLE
Update version of strawberry

### PR DESCRIPTION
**Pull request by patrick91**
This PR updates strawberry to 0.124.0, which improves the support for federation 2. The main change is the support for the link directive, which is also automatically added when opting into federation 2.

I've also updated the schema with support for `override` and fixed a couple of bugs 😊

### CODE REVIEW

File: implementations/strawberry-graphql/server.py [hunk 1]
```diff
@@ -68,14 +68,38 @@ def get_product_dimensions() -> Optional["ProductDimension"]:
...
 
 @strawberry.federation.type(extend=True, keys=["email"])
 class User:
-    email: strawberry.ID = strawberry.federation.field(external=True)
+    email: strawberry.ID = strawberry.federation.field(external=True, override="users")
     name: Optional[str] = strawberry.federation.field(override="users")
     total_products_created: Optional[int] = strawberry.federation.field(external=True)
+    # TODO: the camel casing will be fixed in the next release of Strawberry
+    years_of_employment: int = strawberry.federation.field(external=True)
+    average_products_created_per_year: Optional[int] = strawberry.federation.field(
```
Comment (dariuszkuc) [inline implementations/strawberry-graphql/server.py:86]: I've updated https://github.com/apollographql/apollo-federation-subgraph-compatibility/blob/main/CONTRIBUTORS.md with information how to calculate the field

```
average_products_created_per_year = if (total_products_created != null) {
 round(1.0f * total_products_created /  years_of_employment)
} else {
  null
}
```

Comment (patrick91) [reply implementations/strawberry-graphql/server.py]: Should we add a test that checks for this too?

Comment (dariuszkuc) [reply implementations/strawberry-graphql/server.py]: Unsure how we would test for that -> we can only verify what is the expected return value (which could be hard coded)

**Comment by dariuszkuc** [comment]
Related: https://github.com/apollographql/apollo-federation-subgraph-compatibility/issues/128

**Pull request merged**